cfgov/unprocessed/js/molecules/GlobalSearch.js

```diff
@@ -0,0 +1,171 @@
+    _clearBtnDom = _contentDom.querySelector( clearBtnSelector );
+    var searchIconDom = _contentDom.querySelector( searchIconSelector );
+
+    _triggerDom.addEventListener( 'click', _triggerClicked );
+    _clearBtnDom.addEventListener( 'click', _clearClicked );
+    _searchInputDom.addEventListener( 'keyup', _inputTyped );
+    searchIconDom.addEventListener( 'click', _searchLabelClicked );
+
+    _setClearBtnState( _searchInputDom.value );
+
+    return this;
+  }
+
+  /**
+   * Event handler for when the search icon is clicked in the
+   * expanded state at desktop sizes. Closes the search box.
+   */
+  function _searchLabelClicked() {
+    var currentBreakpoint = breakpointState.get();
+    if ( currentBreakpoint.isBpMED ||
+         currentBreakpoint.isBpLG ||
+         currentBreakpoint.isBpXL ) {
+      collapse();
+    }
+  }
+
+  /**
+   * Event handler for when the search input trigger is clicked,
+   * which opens/closes the search input.
+   */
+  function _triggerClicked() {
+    _isExpanded = !_isExpanded;
+    if ( _isExpanded ) {
+      expand();
+    } else {
+      collapse();
+    }
+  }
+
+  /**
+   * Open the search box.
+   * @returns {Object} An GlobalSearch instance.
+   */
+  function expand() {
+    _isExpanded = true;
+    _triggerDom.setAttribute( 'aria-expanded', 'true' );
+    _contentDom.setAttribute( 'aria-expanded', 'true' );
+    _searchInputDom.focus();
+    _searchInputDom.select();
+
+    return this;
+  }
+
+  /**
+   * Close the search box.
+   * @returns {Object} An GlobalSearch instance.
+   */
+  function collapse() {
+    _isExpanded = false;
+    _triggerDom.setAttribute( 'aria-expanded', 'false' );
+    _contentDom.setAttribute( 'aria-expanded', 'false' );
+
+    return this;
+  }
+
+  /**
+   * @returns {boolean} Whether the search box is expanded or not.
+   *   This can be used to determine if the box is open
+   *   and needs to be closed if an overlapping dialog box is opened.
+   */
+  function isExpanded() {
+    return _isExpanded;
+  }
+
+  /**
+   * Event handler for when the clear input label was clicked.
+   */
+  function _clearClicked() {
+    _searchInputDom.value = _setClearBtnState( '' );
+  }
+
+  /**
+   * Event handler for when the user typed in the input.
+   */
+  function _inputTyped() {
+    _setClearBtnState( _searchInputDom.value );
+  }
+
+  /**
+   * @param {string} value - The input value in the search box.
+   * @returns {string} The input value in the search box.
+   */
+  function _setClearBtnState( value ) {
+    if ( value !== '' ) {
+      _showClearBtn();
+    } else {
+      _hideClearBtn();
+    }
+
+    return value;
+  }
+
+  /**
+   * Add a hidden class to the input search label.
+   * Used when there is no text input.
+   */
+  function _hideClearBtn() {
+    _clearBtnDom.classList.add( 'u-hidden' );
+  }
+
+  /**
+   * Remove a hidden class from the input search label.
+   * Used when there is text input.
+   */
+  function _showClearBtn() {
```

Comment:
Probably out of scope, but we could abstract these two as a `toggleVisibility` utility so that we're not repeating how we hide/show content throughout the project.